app/navigation/navigation_modal.js

```diff
@@ -24,16 +24,20 @@ export default class NavigationModal extends PureComponent {
     }
 
     componentWillReceiveProps(nextProps) {
-        if (this.props.show === nextProps.show) {
+        const nextRoutesLength = nextProps.children.props.navigationState.routes.length;
+        const currentRoutesLength = this.state.children ? this.state.children.props.navigationState.routes.length : 0;
+        const routesDidChange = nextRoutesLength !== currentRoutesLength;
```

Comment:
Is there anywhere that we replace scenes on the navigation stack? We might even be able to just check if `nextProps.children.props.navigationState.routes !== this.state.children.props.navigationState.routes` since it should be immutable in the store.